manifests/resource/location.pp

```diff
@@ -407,8 +407,12 @@
 
   $config_file = "${server_dir}/${server_sanitized}.conf"
 
-  $location_sanitized_tmp = regsubst($location, '\/', '_', 'G')
-  $location_sanitized = regsubst($location_sanitized_tmp, '\\\\', '_', 'G')
+  if !($location =~ /^~/) {
+    $location_sanitized_tmp = regsubst($location, '\/', '_', 'G')
+    $location_sanitized = regsubst($location_sanitized_tmp, '\\\\', '_', 'G')
+  } else {
+    $location_sanitized = $location_sanitized_tmp
```

Comment:
I agree with the proposed changes; that said, maybe I'm being dumb, but can someone point to me where `$location_sanitized` is actually being used in current version of the module? The template only references `$location`, and a recursive grep doesn't turn up other instances of `$location_sanitized`.  Also, I don't think locations would work if we actually turned slashes into underscores even in locations that aren't regexes, so unless it were used for something less direct (like naming fragments), I'm not sure that this code is really doing something that would cause a problem.... Is it the latest version of the module you're seeing this with?

Also, don't these proposed changes do the munging if it _is_ a regex, though (`if !($location =~ /^~/) {`), or am I reading that wrong?

Lastly, not a big deal, but if it's not too much of a hassle to change, I'm trying to switch to what @hunner said is the new convention instead of things like `$foo_real` or `$foo_tmp`, which is 
e.g., (in `server.pp`: `$_ssl_redirect_port = $ssl_redirect_port`). Or maybe just do both substitutions in one line? That seems a little tidier, but maybe too hard to read?

I think having good tests for various scenarios (regex / non regex locations with / without forward and backslash characters) will definitely help make sure we don't miss something or introduce new problems.